### PR TITLE
Enable fwd and varlen_fwd on AMD

### PR DESCRIPTION
**Pull request by micmelesse**
Enable flash_attn_func using amd's triton flash attention kernel.

This enables `flash_attn_func`. There will be follow up prs to enable the forward pass of the other flash attention apis.

we skip building the cuda kernels and use the triton kernel on AMD. This is fine for our fork but upstream will probably require a way to have one wheel.

### CODE REVIEW

File: flash_attn/flash_attn_triton_amd.py [hunk 1]
```diff
@@ -1577,7 +1577,72 @@ def fwd(q,
...
+        dropout_p,
+        softmax_scale,
+        zero_tensors,
+        causal,
+        window_size_left,
+        window_size_right,
+        return_softmax,
+        gen_):
+   
+    print("flash_attn_triton_amd.py::varlen_fwd")
```
Comment (vgokhale) [inline flash_attn/flash_attn_triton_amd.py:1601]: Were these for debug, or did you want them to print out when a user calls this fn?

File: tests/test_flash_attn.py [hunk 2]
```diff
@@ -1181,6 +1182,14 @@ def test_flash_attn_output(
 def test_flash_attn_varlen_output(
     seqlen_q, seqlen_k, d, dropout_p, causal, local, alibi, deterministic, mha_type, dtype, kvpacked, forward_only=True
 ):
+    if is_hip():
+        if dropout_p != 0.0:
+            pytest.skip("Dropout not supported in HIP")
+        
+        # skip all cases where seqlen_q, seqlen_k, or d are not powers of 2
+        if not (is_power_of_2(seqlen_q) and is_power_of_2(seqlen_k) and is_power_of_2(d)):
```
Comment (vgokhale) [inline tests/test_flash_attn.py:1190]: Are these tests failing?

Comment (micmelesse) [reply tests/test_flash_attn.py:1190]: Most of them work, but there are some cases that fail. Also the number of tests is a lot. Just running the tests selected in this subset take like an hour

File: flash_attn/flash_attn_triton_amd.py (hunk 1 above)
Comment (micmelesse) [reply flash_attn/flash_attn_triton_amd.py:1601]: I will clean it up.

**Comment by micmelesse** [comment]
@vgokhale Can I merge this?

**Pull request merged**